packages/web-components/src/components/va-button/va-button.tsx

```diff
@@ -76,17 +76,20 @@ export class VaButton {
   })
   componentLibraryAnalytics: EventEmitter;
 
-  private handleClick = (e: MouseEvent): void => {
-    // TODO: remove console.log
-    console.log(e);
-    if (this.disableAnalytics) return;
-    this.componentLibraryAnalytics.emit({
-      componentName: 'va-button',
-      action: 'click',
-      details: {
-        // TODO: add analytics event details
-      },
-    });
+  private handleClick = (): void => {
+    if (!this.disableAnalytics) {
+      const detail = {
+        componentName: 'va-button',
+        action: 'click',
+        details: {
+          event: 'cta-button-click',
+          type: this.secondary ? 'secondary' : 'primary',
+          label: this.getButtonText(),
+          'background-color': this.secondary ? 'white' : '#0071bb',
```

Comment:
To make this dynamic, we'd probably want to use [getComputedStyle](https://developer.mozilla.org/en-US/docs/Web/API/Window/getComputedStyle) but if that's put in the click handler (where the analytics event is triggered), then the element background color would be different than it's non-clicked state (I have tested this). 

![Screen Shot 2022-11-14 at 10 26 57 AM](https://user-images.githubusercontent.com/872479/201713055-2b050083-2ecb-48c7-8e94-47e316830836.png)
![Screen Shot 2022-11-14 at 10 26 59 AM](https://user-images.githubusercontent.com/872479/201713053-2295855e-e570-41bc-a49e-5c1f83f1e6b1.png)

If we want to get the background color of the non-click state, that means we'd probably want to generate the computed style within a component lifecycle which also means we'd probably want to create a private variable to store it.

So we're talking about adding a dozen additional lines of code to make the color value dynamic just for a piece of analytics data that we don't even know why it's necessary or needed. If we can't verify the reason this value is needed or don't want to hard code a value then I don't think there's enough reason to add any additional code for it and I propose we just remove it from the analytics event entirely.

Reply:
Since it was part of the PR when you picked it up, maybe it would be good if we asked Michelle if it's truly needed or not.

Reply:
@michelle-dooley We're looking at the `va-button` GA event `cta-button-click`. We're not sure if the detail `background-color` is actually necessary for the design system component. I think it might have come out of a conversation that happened in the analytics request ticket here but I don't know the context (if any): https://github.com/department-of-veterans-affairs/va.gov-team/issues/45510#issuecomment-1216867680 

Do you remember if this was a specific request from Katy or is that something that you require on your end? We are considering removing it because we're not sure that the additional code needed to make the color value dynamic is worth the value of collecting the data.

Reply:
Removing `background-color` after [confirming with Michelle](https://github.com/department-of-veterans-affairs/component-library/pull/513#issuecomment-1315799366).